### PR TITLE
executor: replace Executor interface{} with Datum.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

**Comment by coocood** [comment]
@shenli @zimulala PTAL

**Comment by shenli** [comment]
LGTM

File: executor/executor.go [hunk 1]
```diff
@@ -1124,14 +1124,16 @@ func (e *UnionExec) Next() (*Row, error) {
 			for i := range row.Data {
 				// The column value should be casted as the same type of the first select statement in corresponding position
 				rf := e.fields[i]
-				row.Data[i], err = types.Convert(row.Data[i], &rf.Column.FieldType)
+				val := row.Data[i].GetValue()
+				val, err = types.Convert(val, &rf.Column.FieldType)
```
Comment (zimulala) [inline executor/executor.go]: check errors first ?

**Comment by zimulala** [comment]
LGTM

**Pull request merged**